dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/DeploymentProblemTest.java

```diff
@@ -40,39 +35,45 @@ public class DeploymentProblemTest extends FATServletClient {
 
     @BeforeClass
     public static void setup() throws Exception {
-        WebArchive war = ShrinkWrap.create(WebArchive.class, "DeploymentProblemTest.war").addPackage(DuplicateToolErrorTest.class.getPackage());
-        ShrinkHelper.exportDropinAppToServer(server, war, DISABLE_VALIDATION, SERVER_ONLY);
-        server.startServer();
+        String warFileName = "DeploymentProblemTest.war";
+        Package packageName = DuplicateToolErrorTest.class.getPackage();
+        ExpectedAppFailureUtil.setupAndStartServer(server, warFileName, packageName);
     }
 
     @AfterClass
     public static void teardown() throws Exception {
-        server.stopServer("CWWKZ0002E"); // App failed to start
+        ExpectedAppFailureUtil.stopServerIgnoringErrorMessages(server); // ignore app failed to start error
     }
 
+    // This needs to be the first test method in the class
     @Test
-    public void testDuplicateToolDeploymentError() throws Exception {
-        String request = """
-                        {
-                          "jsonrpc": "2.0",
-                          "id": 1,
-                          "method": "tools/list",
-                          "params": {
-                            "cursor": "optional-cursor-value"
-                          }
-                        }
-                        """;
+    public void testAppFailedToStart() throws Exception {
+        assertTrue(ExpectedAppFailureUtil.appHasFailed(server));
+    }
 
-        boolean deploymentErrorOccured = false;
+    @Test
+    public void testDuplicateToolDeploymentError() throws Exception {
+        List<String> expectedErrorList = List.of("io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.DuplicateToolErrorTest.bob()",
+                                                 "io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.DuplicateToolErrorTest.duplicateBob()",
```

Comment:
You've removed a check that was present in the original test and I think is easy to add back in here.

```suggestion
        List<String> expectedErrorList = List.of("More than one MCP tool has the same name",
                                                 "io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.DuplicateToolErrorTest.bob()",
                                                 "io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.DuplicateToolErrorTest.duplicateBob()",
```